jni/events/sampleevent.cpp

```diff
@@ -200,13 +202,23 @@ void SampleEvent::mixBuffer( AudioBuffer* outputBuffer, int bufferPos, int minBu
     // if we have a range length that is unequal to the total sample duration, read from the range
     // otherwise invoke the base mixBuffer method
 
-    if ( _bufferRangeLength != _sampleLength )
+    if ( _useBufferRange )
         getBufferForRange( outputBuffer, bufferPos );
```

Comment:
Range based playback must be specifically requested (by use of the setter or one of the bufferRange setters)